packages/sources/svelte/src/components/navigation/VtmnSearch/VtmnSearch.svelte

```diff
@@ -0,0 +1,109 @@
+<script>
+  import { cn } from '../../../utils/classnames';
+
+  import { VtmnButton } from '@vtmn/svelte';
+  import { createEventDispatcher } from 'svelte';
+  import { VARIANTS, SIZES } from './enums';
+
+  /** @restProps { button } */
+
+  /**
+   * The variant of the button.
+   * @type {'default' | 'ghost' | 'persistent' | 'on-content'}
+   * @defaultValue 'default'
+   */
+  export let variant = VARIANTS.DEFAULT;
+
+  /**
+   * @type {boolean} disabled.
+   * @defaultValue false
+   */
+  export let disabled = false;
+
+  /**
+   * @type {'small' | 'medium'}
+   * @defaultValue 'medium'
+   */
+  export let size = SIZES.MEDIUM;
+
+  /**
+   * The value of the input
+   * @type {string}
+   */
+  export let value;
+
+  /**
+   * Custom classes to apply to the component.
+   * @type {string}
+   */
+  export let className;
+
+  const dispatch = createEventDispatcher();
+
+  const resetInputValue = () => {
+    value = '';
+  };
+
+  const onSearch = () => {
+    dispatch('search', {
+      text: value,
+    });
+  };
+
+  $: componentClass = cn(
+    'vtmn-search',
+    variant && `vtmn-search_variant--${variant}`,
+    size && `vtmn-search_size--${size}`,
+    className,
+  );
+</script>
+
+<form
+  class={componentClass}
+  action=""
+  role="search"
+  aria-controls="search-result"
+  on:submit|preventDefault={onSearch}
+>
```

Comment:
I don't kniw if it is a good thing to embed the input in a form, I would rather let it standalone and let the developers do this themselves.
@lauthieb ?

Reply:
I will delete the form because I went to see in the other frameworks like material ui and the form tag is not directly in this type of components